### PR TITLE
feat(parseEditorState): adding support for parsed JSON in addition to stringified JSON

### LINKED ISSUE #2052: Feature: expose `parseEditorState` function that accepts already parsed JSON.
My datastore stores JSON natively so it feels funny to have to re-stringify the value just to have it parsed back by `editor.parseEditorState()`. Could we extend `editor.parseEditorState()` to accept an object of same shape? Or just expose the `parseEditorState` utility function that it internally calls?

## Alternative
Just re-stringify the obj before sending to the method.

### PR DESCRIPTION
**Pull request by nicksrandall**
Fixes #2052

### CODE REVIEW

**Comment by fantactuka** [comment]
Mind adding a simple test?

**Comment by nicksrandall** [comment]
@fantactuka Thanks for the help. A simple test has been added but I'm not sure if it is enough. Let me know.

File: packages/lexical/src/__tests__/unit/LexicalEditor.test.js [hunk 1]
```diff
@@ -1062,6 +1082,13 @@ describe('LexicalEditor tests', () => {
         expect(parsedSelection.anchor.key).toEqual(parsedText.__key);
         expect(parsedSelection.focus.key).toEqual(parsedText.__key);
       });
+
+      it('Accepts already parsed editor state', async () => {
+        expect(parsedSelection.anchor.key).not.toEqual(originalText.__key);
+        expect(parsedSelection.focus.key).not.toEqual(originalText.__key);
+        expect(parsedSelection.anchor.key).toEqual(parsedText.__key);
+        expect(parsedSelection.focus.key).toEqual(parsedText.__key);
+      });
```
Comment (fantactuka) [inline packages/lexical/src/__tests__/unit/LexicalEditor.test.js]: You can move these assertions inside the `parses parsed JSON` test, as now they are executed for `range selection` test case above

**Review by fantactuka — APPROVED** [review]
Looks good to me, thank you for contributing!

**Pull request merged**